students/lauraannf/Lesson03/mailroom.py

```diff
@@ -4,53 +4,73 @@
-          'Laura F'.format(Name,Donation))
-def reportfunc():
+          'Laura F'.format(name, donation))
+
+
+def create_email():
+    while True:
+        name = input("Type full donor name, or ''list'' for choices?>")
+        if name.lower() == 'list':
+            print(donor_list[0::2])
+        elif name not in donor_list:
+            donor_list.append(name)
+            donation = input("Donation Amount?>")
+            donation = float(donation)
+            donor_list.append([donation])
+            write_email(name, donation)
+            break
+        else:
+            indx = donor_list.index(name)
+            donation = input("Donation Amount?>")
+            donation = float(donation)
+            donor_list[indx+1].append(donation)
+            write_email(name, donation)
+            break
+
+
+def create_report():
     print('-------List of Donors-------')
-    print('{:<20}{:<20}{:<20}{:<20}'.format('Donor Name', 'Total Donated', '# of Donations', 'Average Donation'))
+    print('{:<20}{:<20}{:<20}{:<20}'.format('Donor Name', 'Total Donated',
+                                            '# of donations',
+                                            'Average donation'))
     print('-----------------   '*4)
-    for it in range(0,len(DonorList),2):
-        print('{:<20}${:<20.2f}{:<20d}${:<20.2f}'.format(DonorList[it], sum(DonorList[it+1]), len(DonorList[it+1]),  sum(DonorList[it+1])/len(DonorList[it+1])))              
-    
+    for it in range(0, len(donor_list), 2):
+        print('{:<20}${:<20.2f}{:<20d}${:<20.2f}'.format(donor_list[it],
+                                                         sum(donor_list[it+1]),
+                                                         len(donor_list[it+1]),
+                                                         sum(donor_list[it+1])
+                                                         / len(donor_list[it+1]
+                                                               )))
+
+
 def main():
     print('----------Mailroom------------')
-    Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
-    while Task not in ('1','2','3'):
-        Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
-    while Task !='3':  
-        if Task == '1':
-            Name = input("Type full donor name, or ''list'' for choices?>")
-            namefunc(Name)
-            Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
-        elif Task == '2':
-            reportfunc()
-            Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
+    while True:
+        task = input("Choose an action: [1] Send a Thank You; [2] Create a \
+Report; [3] Quit>")
+        if task == '1':
+            create_email()
+        elif task == '2':
+            create_report()
+        elif task == '3':
+            break
+        else:
+            continue
```

Comment:
no need for continue here as it s redundant, your loop will continue looping on its own.